worldedit-core/src/main/java/com/fastasyncworldedit/core/internal/simd/VectorizedMask.java

```diff
@@ -0,0 +1,39 @@
+package com.fastasyncworldedit.core.internal.simd;
+
+import com.fastasyncworldedit.core.queue.IChunk;
+import com.fastasyncworldedit.core.queue.IChunkGet;
+import com.fastasyncworldedit.core.queue.IChunkSet;
+import jdk.incubator.vector.ShortVector;
+import jdk.incubator.vector.VectorMask;
+import jdk.incubator.vector.VectorSpecies;
+
+public interface VectorizedMask {
+
+    default void processChunks(IChunk chunk, IChunkGet get, IChunkSet set) {
+        for (int layer = get.getMinSectionPosition(); layer <= get.getMaxSectionPosition(); layer++) {
+            final char[] sectionSet = set.loadIfPresent(layer);
+            if (sectionSet == null) {
+                continue;
+            }
+            final char[] sectionGet = get.load(layer);
+            processSection(layer, sectionSet, sectionGet);
+        }
+    }
+
+    default void processSection(int layer, char[] set, char[] get) {
+        final VectorSpecies<Short> species = ShortVector.SPECIES_PREFERRED;
+        for (int i = 0; i < set.length; i += species.length()) {
+            ShortVector vectorSet = ShortVector.fromCharArray(species, set, i);
```

Comment:
Probably worth adding a comment here (and where-ever else) that it's a good assumption that the vector size supported will divide into 4096. I see no reason it should ever not, but the set and get arrays are technically not locked to 4096